fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -126,4 +133,18 @@ public CollectionData calculateLoanCollectionData(final Long loanId) {
         return collectionData;
     }
 
+    @Override
+    public Collection<LoanInstallmentDelinquencyTagData> retrieveLoanInstallmentsCurrentDelinquencyTag(Long loanId) {
+
+        List<LoanInstallmentDelinquencyTagData> loanInstallmentsDelinquencyData = new ArrayList<>();
+        List<LoanInstallmentDelinquencyTag> loanInstallmentsCurrentDelinquency = repositoryLoanInstallmentDelinquencyTag
+                .findByLoanId(loanId);
+        for (LoanInstallmentDelinquencyTag installmentDelinquencyTag : loanInstallmentsCurrentDelinquency) {
```

Comment:
Still loading the loan for no reason. Please fetch the required `loanInstallmentsDelinquencyData` via the repository using JPA projections:

https://www.baeldung.com/spring-data-jpa-projections

Reply:
done. Fetching only Delinquency Range and  outstanding amount for installments